src/Transport/Config/ExtensionPoints/AzureServiceBusNamespacePartitioningSettings.cs

```diff
@@ -29,7 +29,7 @@ public AzureServiceBusNamespacePartitioningSettings(SettingsHolder settings)
             return this;
         }
 
-        public AzureServiceBusNamespacePartitioningSettings AddNamespace(string name, string connectionString)
+        public AzureServiceBusNamespacePartitioningSettings AddNamespace(string name, string connectionString, NamespacePurpose purpose = NamespacePurpose.Partitioning)
```

Comment:
`NamespaceName` is a useful info only if a user enables `UseNamespaceNameInsteadOfConnectionString`. Why not put the `namespace name` configuration inside that config api? 
Here the same concept I suggested for ASQ: https://github.com/Particular/NServiceBus.AzureStorageQueues/blob/release-7.0.0/src/Transport/Config/Extensions/AzureStorageTransportAddressingExtensions.cs#L14

We could add `purpose` there, and use partitioning config API to setup only partitioning related info